mbed_tools_ci_scripts/license_files.py

```diff
@@ -15,54 +15,71 @@
 from pathlib import Path
 
 logger = logging.getLogger(__name__)
 
-LICENCE_HEADER_TEMPLATE = """Copyright (c){date}, {author}. All rights reserved.
+LICENCE_HEADER_TEMPLATE = """Copyright (c) {date}, {author}. All rights reserved.
 SPDX-License-Identifier: {licence_identifier}
 """
 
 
-class CopyrightNotifierToolWrapper:
-    """Wrapper over the licenseheaders tool.
+class LicenceHeader:
+    """Tool adding a licence header at the top of every source file.
 
-    See https://github.com/johann-petrak/licenseheader for more details about the tool in use.
-    copyright (https://github.com/knipknap/copyright) was first considered but
-    comprises quite a few bugs and does not seem active anymore.
+    Wrapper over the [licenseheaders tool](https://github.com/johann-petrak/licenseheaders).
     """
 
+    # copyright (https://github.com/knipknap/copyright) was first considered but
+    # comprises quite a few bugs and does not seem active anymore.
     def run(self, verbose_count: int) -> None:
         """Puts a copyright notice at the top of every source file."""
-        template_string = LICENCE_HEADER_TEMPLATE.format(
-            licence_identifier=configuration.get_value(ConfigurationVariable.FILE_LICENCE_IDENTIFIER),
-            author="${owner}",
-            date=" ${years}",
-        )
+        template_string = self._generate_header_template()
         with tempfile.NamedTemporaryFile(suffix=".tmpl", delete=False) as template_file:
             template_file_path = Path(template_file.name)
             logger.debug(f"Creates template file in {str(template_file_path)}")
             template_file.write(template_string.encode("utf8"))
             template_file.close()
             copyright_config = self.get_tool_config(template_file_path)
-            self.run_licensehearders(copyright_config, verbose_count)
+            self._call_licensehearders(copyright_config, verbose_count)
 
-    def run_licensehearders(self, config: dict, verbose_count: int) -> None:
+    def _generate_header_template(self) -> str:
+        """Generates the header template which is put at the top of source files."""
+        return LICENCE_HEADER_TEMPLATE.format(
+            licence_identifier=configuration.get_value(ConfigurationVariable.FILE_LICENCE_IDENTIFIER),
+            author="${owner}",
+            date="${years}",
+        )
+
+    def _call_licensehearders(self, config: dict, verbose_count: int) -> None:
         """Runs licenseheaders tool."""
         args = ["licenseheaders"]
-        args.extend([item for pair in {f"--{k}": str(v) for (k, v) in config.items()}.items() for item in pair])
+        args_dict = {f"--{k}": str(v) for (k, v) in config.items()}
+        args.extend(flatten_dictionary(args_dict))
         if verbose_count > 0:
             args.append(f"-{''.join(['v'] * verbose_count)}")
         subprocess.check_call(args)
 
+    def _determines_copyright_dates(self) -> str:
```

Comment:
Should this be a `staticmethod` or better yet, a free function? `self` isn't used here.